### ISSUE TITLE
[FEATURE] Easily disable all animations on global config

### ISSUE THREAD

**Issue by basselalsayed** [framer/motion#1604]
**Is your feature request related to a problem? Please describe.**
No

**Describe the solution you'd like**
This is more of a style choice, in my own usage I implemented something like the following using context:
```
const DisableableMotionDiv: ForwardRefComponent<HTMLDivElement, HTMLMotionProps<'div'>> = forwardRef(
  props => {
    const { animationsEnabled } = useUiContext();

    let newProps = props;

    if (!animationsEnabled) {
      newProps = {
        ...props,
        animate: undefined,
        transition: undefined,
        initial: undefined,
        variants: undefined,
        whileHover: undefined,
        whileDrag: undefined,
        whileFocus: undefined,
        whileInView: undefined,
        whileTap: undefined,
      };
    }

    return <motion.div {...newProps} />;
  },
);
```

**Describe alternatives you've considered**
* a global window variable would be easy for devs to use but I don't think changes would be reconciled by react.
* An optional `MotionContext` component
* Extending the `MotionConfig` component to disable all animations

**Additional context**
* Happy to code out a solution and contribute

### Does Framer Motion support Preact?
It might work with Preact, but we don't explicitly support it, so if it does currently work you'd be using it at your own risk.

**Comment by stutrek** [comment]
I would also like this feature. I am trying to use snapshot testing, which takes a screenshot and compares it to a known good state. Because of opacity transitions, we get false negatives nearly every run.

**Comment by sanpoChew** [comment]
I think this is the same as https://github.com/framer/motion/issues/1160 which looks like it might be worked on soon

**Comment by mattgperry** [comment]
Implemented in 10.17.0


```
import { MotionGlobalConfig } from "framer-motion"

MotionGlobalConfig.skipAnimations = true
```

**Comment by lukpsaxo** [comment]
It took some debugging to work out why it wasn't working. If you use motion/react, framer-motion is bundled so you need to do this:

```
import { MotionGlobalConfig } from 'motion/react';

MotionGlobalConfig.skipAnimations = true;
```

Hope it helps someone!
